### ISSUE TITLE
Approximate time to finish prepare_mapped_reads

### ISSUE THREAD

**Issue by marcpaga** [nanoporetech/taiyaki#92]
Hi,

I am running the `prepare_mapped_reads.py` script as described in the walkthrough. I tried to run it with 32 threads but it didn't finish in 12h. I have 40.000 reads from the data provided in the walkthrough. Could someone provide an approximation on how much time it should take to run? I am trying to see if there is something wrong on my end before running it again. 

Thanks!

**Comment by tmassingham-ont** [comment]
Hello.   I've started a run to get sample timings.  Will update soon.

**Comment by marcpaga** [comment]
Many thanks @tmassingham-ont! I am looking forward to seeing the results.

**Comment by tmassingham-ont** [comment]
TL;DR: 5 hours, if you have sufficient RAM and CPU.

Did you see a progress indicator (rows of coloured dots) at all?

Using the commands in the walk-through, I get the following times on a (rather old) E5-2650 v4 with 24 cores; base calling performed on RTX 2080.  Times rounded.
```
  11m40  guppy_basecaller
   3m40  minimap2
   3m20  get_refs_from_sam.py
   5m00  generate_per_read_params.py
4h54m10  prepare_mapped_reads.py
```

One potential issue is that all the output is formed in RAM and then written at once.  A machine with insufficient resources might suffer and a major slow-down would be apparent.  In these circumstances, the data could be prepared in batches by splitting up `input_strand_list` then merged using the `misc/merge_mappedsignalfiles.py` utility (this can also be used to split the mapping over several machines).

When preparing your own data, it is important to know that the preparation scales very badly (quadratic) with the length of reads and most of the runtime will be spent on a few reads. Setting the `--max_read_length` flag to filter out long reads is helpful.

**Comment by marcpaga** [comment]
Thanks for the detailed answer! This is very helpful. I think the problem is on my end, while the CPU and RAM should not be a problem, hard drive access is a bit slow, so reading all those reads might be the issue here.
Just to add a bit in case it is useful for someone else, I had already filtered long reads. They either had max 100k raw datapoints or 10k basecalls. So I think it was a hard drive issue.
Many thanks again!
